packages/nextjs/components/HooksExample.tsx

```diff
@@ -0,0 +1,192 @@
+"use client";
+
+import { useState } from "react";
+import { useScaffoldContract } from "~~/hooks/scaffold-stark/useScaffoldContract";
+import {
+  createContractCall,
+  useScaffoldMultiWriteContract,
+} from "~~/hooks/scaffold-stark/useScaffoldMultiWriteContract";
+import { useTargetNetwork } from "~~/hooks/scaffold-stark/useTargetNetwork";
+import { useScaffoldReadContract } from "~~/hooks/scaffold-stark/useScaffoldReadContract";
+import { useScaffoldEventHistory } from "~~/hooks/scaffold-stark/useScaffoldEventHistory";
+import { ContractName } from "~~/utils/scaffold-stark/contract";
+import { useScaffoldWriteContract } from "~~/hooks/scaffold-stark/useScaffoldWriteContract";
+
+const HooksExample: React.FC = () => {
+  const [contractName, setContractName] = useState<ContractName>("YourContract");
+  const [newGreeting, setNewGreeting] = useState<string>("");
+  const [amountEth, setAmountEth] = useState<string>("");
+  const [fromBlock, setFromBlock] = useState<bigint>(0n);
+
+  const { data: contract, isLoading: isContractLoading } = useScaffoldContract({
+    contractName: contractName as ContractName,
+  });
+
+  const { targetNetwork } = useTargetNetwork();
+
+  const { writeAsync: setGreetingMulti } = useScaffoldMultiWriteContract({
+    calls: [
+      createContractCall(
+        contractName as ContractName,
+        "set_gretting",
+        [newGreeting, BigInt(amountEth)],
```

Comment:
the correct hook to use here is the `useScaffoldNultiWrite` , because eth has to be approved to spender first , then call `setGreeting()`